tutorials/containers/deploying-storageos-on-kubernetes/deploying-storageos-on-kubernetes.md

```diff
@@ -0,0 +1,186 @@
+type: "kubernetes.io/storageos"
+data:
+  apiUsername: dWJ1bnR1
+  apiPassword: dWJ1bnR1
+               ^^^^^^^^
+ echo ‘ubuntu’ | base64
+END
+```
+
+Now that you have the StorageOS initial account configured you can go ahead and install StorageOS, this step is important and if not configured properly you may get an error similar to this:
+
+```
+storageos-daemonset-qpwfn             0/1     Init:Error              11         14m
+storageos-daemonset-qpwfn             0/1     PodInitializing         0          14m
+```
+
+## Install StorageOS
+
+To configure StorageOS we can use the YAML from the StorageOS docs, for this to work you will require your cluster policy to allow privileged.
+
+```
+apiVersion: "storageos.com/v1"
+kind: StorageOSCluster
+metadata:
+  name: "example-storageos"
+  namespace: "storageos-operator"
+spec:
+  secretRefName: "storageos-api" # Reference the Secret created in the previous step
+  secretRefNamespace: "storageos-operator"  # Namespace of the Secret
+  k8sDistro: "kubernetes"
+  images:
+    nodeContainer: "storageos/node:1.4.0" # StorageOS version
+  resources:
+    requests:
+    memory: "512Mi"
+```
+
+This will trigger 3 daemonset pods to install and 1 scheduler pod, you can check the installation status with:
+
+```
+$ kubectl -n storageos get pods -w
+```
+
+## Mounting a volume
+
+Ok, now you can create a Persistent Volume Claim:
+
+```
+kubectl create -f - <<END
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: my-first-storage-os-volume
+  annotations:
+    volume.beta.kubernetes.io/storage-class: fast
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+END
+```
+
+And check its status:
+
+```
+$ kubectl -n storageos get pods -w
+```
+
+You should get something like this:
+
+```
+[My-first…] Bound pvc-72866c11-fe5b-40e4-a156-6bcfe7ade53e 5Gi RWO fast 4m14s
+```
+
+## Deploying on a workload
+
+Now you can create a pod on our Charmed Kubernetes cluster and try out the newly bound PVC, this can be generalised for any deployment on Kubernetes.
+
+```
+kubectl -f - <<END
+apiVersion: v1
+kind: Pod
+metadata:
+  name: storageos-ubuntu
+spec:
+  containers:
+    - name: ubuntu
+      image: ubuntu:bionic
+      command: ["/bin/sleep"]
+      args: [ "3600" ]
+      volumeMounts:
+        - mountPath: /mnt
+          name: my-mount
+  volumes:
+    - name: my-mount
+      persistentVolumeClaim:
+        claimName: my-first-storage-os-volume
+END
+```
+Check its status:
+```
+$ kc get pods
+```
+You should see something like this:
+```
+NAME               READY   STATUS    RESTARTS   AGE
+storageos-ubuntu   1/1     Running   0          2m43s
+```
+
+So now you can write and read from our new mounted volume:
+```
+$ kubectl exec -it storageos-ubuntu -- bash
+```
+Write something to the new volume:
+```
+root@storageos-ubuntu:/# echo "Ubuntu + StorageOS!" > /mnt/myfile
+```
+And read it!
+```
+root@storageos-ubuntu:/# cat /mnt/myfile
+```
```

Comment:
The summary/conclusions section seems to be missing. Please add one based on the template you used for your previous tutorial. Don't forget to attach some interesting links:

https://microk8s.io/
https://jaas.ai/kubernetes
https://github.com/ubuntu/microk8s/issues
https://ubuntu.com/kubernetes/contact-us